src/models/tools/table/table-content.ts

```diff
@@ -335,9 +336,21 @@ export const TableContentModel = types
           const parser = new Parser();
           const parsedExpression = parser.parse(expression);
           for (let i = 0; i < attr.values.length; i++) {
-            const xVal = xAttr.value(i) as number;
-            const expressionVal = parsedExpression.evaluate({[kSerializedXKey]: xVal});
-            attr.setValue(i, isFinite(expressionVal) ? expressionVal : "");
+            const xVal = xAttr.value(i) as number | string;
+            if (xVal == null || xVal === "") {
+              attr.setValue(i, undefined);
+            } else {
+              const expressionVal = parsedExpression.evaluate({[kSerializedXKey]: xVal});
+              attr.setValue(i, isFinite(expressionVal) ? expressionVal : kErrorString);
```

Comment:
I would recommend storing the value as a NaN internally and then handling the `#ERR` string as a display issue in the table's valueGetter. We only have one level of formula evaluation at this point, but down the road we'll have multiple levels and NaNs propagate in predictable ways whereas the `#ERR` string won't necessarily do so.